### ISSUE TITLE
DB corrupted: Corruption: block checksum mismatch

### ISSUE THREAD

**Issue by donewebb** [paritytech/parity-ethereum#9867]
New to mining - started the sync 3 days ago after it looked to be complete I started ethminer pointing to the Parity which appeared to be getting work for about 24 hours before the crash . when I checked today Parity had died with an error: below , after I restarted I received this message. When Parity died it provided this link and mentioned open a bug ticket, I suppose now the only thing to do is delete everything and start over because the restart doesn't correct the issue?
If there is anything I am required to provide just let me know how to help. I believe I downloaded the current versions of Parity and ethminer version numbers are below as well as windows 10 version.

RESTART:

Microsoft Windows [Version 10.0.17134.345]

C:\Mining\Parity>parity --author XXXXX
2018-11-05 07:16:23  Starting Parity-Ethereum/v2.0.9-stable-09f77575d-20181028/x86_64-windows-msvc/rustc1.29.1
2018-11-05 07:16:23  Keys path C:\Users\XXXXXX\AppData\Roaming\Parity\Ethereum\keys\ethereum
2018-11-05 07:16:23  DB path C:\Users\XXXXXX\AppData\Local\Parity\Ethereum\chains\ethereum\db\906a34e69aec8c0d
2018-11-05 07:16:23  State DB configuration: fast
2018-11-05 07:16:23  Operating mode: active
2018-11-05 07:16:23  DB has been previously marked as corrupted, attempting repair
2018-11-05 07:19:30  DB corrupted: Invalid argument: You have to open all column families. Column families not opened: col0, col4, col3, col2, col6, col1, col5, attempting repair
Failed to open database Custom { kind: Other, error: StringError("Received null column family handle from DB.") }

ERROR:

2018-11-04 23:47:38    #2628851   21/25 peers    359 KiB chain  114 MiB db  0 bytes queue   38 MiB sync  RPC:  0 conn,    1 req/s, 8741 µs
2018-11-04 23:47:49  Imported #6646274 0xa993…bf31 (122 txs, 7.98 Mgas, 1113 ms, 22.28 KiB) + another 1 block(s) containing 244 tx(s)
2018-11-04 23:47:54  Imported #6646275 0xf990…27ee (8 txs, 8.00 Mgas, 53 ms, 1.48 KiB)
2018-11-04 23:48:08  Imported #6646276 0x2303…a358 (124 txs, 7.99 Mgas, 339 ms, 20.18 KiB)
2018-11-04 23:48:08    #2639206   19/25 peers      2 MiB chain  114 MiB db  0 bytes queue   23 MiB sync  RPC:  0 conn,    1 req/s, 4553 µs
2018-11-04 23:48:23  Imported #6646277 0xcf79…6856 (156 txs, 8.00 Mgas, 194 ms, 23.22 KiB)
2018-11-04 23:48:23  DB corrupted: Corruption: block checksum mismatch: expected 2324675200, got 2290660627  in C:\Users\XXXXXX\AppData\Local\Parity\Ethereum\chains\ethereum\db\906a34e69aec8c0d\overlayrecent\db/029161.sst offset 64044496 size 12935. Repair will be triggered on next restart

====================

stack backtrace:
   0:     0x7ff7bbb1daa3 - hid_error
   1:     0x7ff7bbb1d71e - hid_error
   2:     0x7ff7bbb1d79d - hid_error
   3:     0x7ff7bbb1c51e - hid_error
   4:     0x7ff7ba5e57f2 - <no info>
   5:     0x7ff7bbb783f2 - hid_error
   6:     0x7ff7bbb77f54 - hid_error
   7:     0x7ff7bbb77e39 - hid_error
   8:     0x7ff7bbb92c2c - hid_error
   9:     0x7ff7bb3da03c - <no info>
  10:     0x7ff7bb2c3900 - <no info>
  11:     0x7ff7bb0457de - <no info>
  12:     0x7ff7bb48a90c - <no info>
  13:     0x7ff7bb2f5669 - <no info>
  14:     0x7ff7bb2f6d35 - <no info>
  15:     0x7ff7bb40cfdd - <no info>
  16:     0x7ff7bb533091 - <no info>
  17:     0x7ff7bbb8eb52 - hid_error
  18:     0x7ff7bb36ef52 - <no info>
  19:     0x7ff7bbb771e2 - hid_error
  20:     0x7ff7bbb72839 - hid_error
  21:     0x7ffe17903034 - BaseThreadInitThunk

Thread 'Verifier #1' panicked at 'DB flush failed.: Custom { kind: Other, error: StringError("Corruption: block checksum mismatch: expected 2324675200, got 2290660627  in C:\\Users\\XXXXXX\\AppData\\Local\\Parity\\Ethereum\\chains\\ethereum\\db\\906a34e69aec8c0d\\overlayrecent\\db/029161.sst offset 64044496 size 12935") }', libcore\result.rs:945

This is a bug. Please report it at:

    https://github.com/paritytech/parity/issues/new

2018-11-04 23:48:23  Finishing work, please wait...
2018-11-04 23:48:25  DB corrupted: Corruption: block checksum mismatch: expected 2324675200, got 2290660627  in C:\Users\XXXXXX\AppData\Local\Parity\Ethereum\chains\ethereum\db\906a34e69aec8c0d\overlayrecent\db/029161.sst offset 64044496 size 12935. Repair will be triggered on next restart

====================

stack backtrace:
   0:     0x7ff7bbb1daa3 - hid_error
   1:     0x7ff7bbb1d71e - hid_error
   2:     0x7ff7bbb1d79d - hid_error
   3:     0x7ff7bbb1c51e - hid_error
   4:     0x7ff7ba5e57f2 - <no info>
   5:     0x7ff7bbb783f2 - hid_error
   6:     0x7ff7bbb77f54 - hid_error
   7:     0x7ff7bbb77e39 - hid_error
   8:     0x7ff7bbb92c2c - hid_error
   9:     0x7ff7ba759d34 - <no info>
  10:     0x7ff7ba69a32a - <no info>
  11:     0x7ff7ba7b679c - <no info>
  12:     0x7ff7ba7b4fd3 - <no info>
  13:     0x7ff7ba79cc6c - <no info>
  14:     0x7ff7ba7a8a46 - <no info>
  15:     0x7ff7bafc6923 - <no info>
  16:     0x7ff7bb8009f7 - hid_error
  17:     0x7ff7ba7a77f4 - <no info>
  18:     0x7ff7ba6486ee - <no info>
  19:     0x7ff7ba643fdc - <no info>
  20:     0x7ff7bb851568 - hid_error
  21:     0x7ff7bb84d4fe - hid_error
  22:     0x7ff7bb84b86c - hid_error
  23:     0x7ff7bb84e86f - hid_error
  24:     0x7ff7bbb8eb52 - hid_error
  25:     0x7ff7bb84df42 - hid_error
  26:     0x7ff7bbb771e2 - hid_error
  27:     0x7ff7bbb72839 - hid_error
  28:     0x7ffe17903034 - BaseThreadInitThunk

Thread '<unnamed>' panicked at 'Propagating verifier thread panic on shutdown: Any', libcore\result.rs:945

This is a bug. Please report it at:

    https://github.com/paritytech/parity/issues/new

2018-11-04 23:48:26  DB corrupted: Corruption: block checksum mismatch: expected 2324675200, got 2290660627  in C:\Users\XXXXXX\AppData\Local\Parity\Ethereum\chains\ethereum\db\906a34e69aec8c0d\overlayrecent\db/029161.sst offset 64044496 size 12935. Repair will be triggered on next restart

**Comment by joshua-mir** [comment]
Yes this does look like something you'll need to sync from scratch again for.

You should run `parity db kill` before restarting your client to wipe the database.

With this many database corruption issues, I should probably ask what kind of hardware you are running your node on? A “block checksum mismatch“ (your first error) shouldn't be happening unless the data is being corrupted *on disk*

**Comment by donewebb** [comment]
new desktop after many years last month.

64bit Windows 10 Pro
Threadripper 2950X  - 32 GBs ram
2 Vega 64 GPUs 8GBs ram each
1 TB SSD
2 TB SATA
everything is set to run the primary SSD

I have been using for mainly gaming without any issue, really haven't had any issues out of it at all works great.
However I wanted to see if I could mine some crypto also, pretty late to the game but I thought it would be an experience , which lasted about 3 days.

Thanks for your advice.

**Comment by joshua-mir** [comment]
Definitely good enough on paper to run a node and probably mine in a pool @donewebb 

However I should *probably* let you know that:
1) you aren't going to get anywhere "solo mining" (as opposed to pool mining) without significant investment in hardware.
2) you don't actually need to run a node yourself to mine in a pool, the pool operator runs the node for you and sends you work.

My guess as to the cause of this issue:
You have alot of RAM. That's great for gaming, but chances are that it isn't ECC enabled. This isn't an issue for gaming (doesn't need error-free memory), or mining (astronomically unlikely to be a problem), but parity can and will use as much memory as possible unless you tell it otherwise, and if any of it gets corrupted by a single bit, it will crash, as you can see. 

Closing as duplicate of #9677, maybe run a memtest and check your SSD for signs of aging just to be safe, but otherwise find a friendly mining pool and follow their instructions to mine, which very likely won't involve setting up a full node.

**Comment by donewebb** [comment]
thanks for the information. I did run a health check against the SSD , at least it reported 100% Healthy. the ram is ADATA XPG Z1 3000 DDR4. 

I read the earlier issue and the only difference was that I did complete syncing the first time without issue.  I started ethminer  it and had been running 2 days before the error appeared. 

Anyway thanks for your help and since it doesn't take to long to completely sync again I am trying to see if the issue appears then I may need to try your other suggestions. I have 2 more RX 580 cards that I could also include. 
I see what you mean about joining a pool I first started with geth and once it was all synced I tried to use the ethminer and noticed unless I was using parity I had to provide a pool address.
but im pretty new at this , I could have been doing something incorrect.
